src/ui/public/directives/filter_invert.js

```diff
@@ -0,0 +1,49 @@
+import _ from 'lodash';
+import $ from 'jquery';
+import uiModules from 'ui/modules';
+import invertFilterTemplate from 'ui/directives/partials/filter_invert.html';
+let module = uiModules.get('kibana');
+
+module.directive('filterInvert', function ($compile) {
+  return {
+    restrict: 'A',
+    link: function ($scope, $elem) {
+      /* from modernizer
+       https://raw.githubusercontent.com/Modernizr/Modernizr/master/feature-detects/css/filters.js
+       */
+      function isFilterSupported() {
+        const el = document.createElement('a');
+        el.style.cssText = '-o-filter:blur(2px); -ms-filter:blur(2px); -moz-filter:blur(2px); -webkit-filter:blur(2px); filter:blur(2px);';
+        return !!el.style.length && ((document.documentMode === undefined || document.documentMode > 9));
+      }
+
+      function invert($elem) {
+        $scope.opts = {
+          idHash: (Math.random().toString(36) + '00000000000000000').slice(2, 7),
```

Comment:
Not that I know of, but does this really need to be random? As far as I can tell, IDs defined inside the SVGs `defs` element are scoped to the individual SVG, so this can probably just be hard coded?